Package frenetic.5.0.5

jnfoster opened the pull request:
### `frenetic.5.0.5`
The Frenetic Programming Language and Runtime System



---
* Homepage: http://frenetic-lang.org
* Source repo: git+https://github.com/frenetic-lang/frenetic.git
* Bug tracker: https://github.com/frenetic-lang/frenetic/issues

---
:camel: Pull-request generated by opam-publish v2.1.0

mseri commented:
All builds are failing with
```
#       ocamlc src/syntax/.frenetic_ppx.objs/byte/frenetic_ppx__Ppx.{cmi,cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.12/bin/ocamlc.opt -w -40 -g -bin-annot -I src/syntax/.frenetic_ppx.objs/byte -I /home/opam/.opam/4.12/lib/astring -I /home/opam/.opam/4.12/lib/base -I /home/opam/.opam/4.12/lib/base/base_internalhash_types -I /home/opam/.opam/4.12/lib/base/caml -I /home/opam/.opam/4.12/lib/base/md5 -I /home/opam/.opam/4.12/lib/base/shadow_stdlib -I /home/opam/.opam/4.12/lib/base64 -I /home/opam/.opam/4.12/lib/base_bigstring -I /home/opam/.opam/4.12/lib/base_quickcheck -I /home/opam/.opam/4.12/lib/bigarray-compat -I /home/opam/.opam/4.12/lib/bin_prot -I /home/opam/.opam/4.12/lib/bin_prot/shape -I /home/opam/.opam/4.12/lib/biniou -I /home/opam/.opam/4.12/lib/bytes -I /home/opam/.opam/4.12/lib/core -I /home/opam/.opam/4.12/lib/core/error_checking_mutex -I /home/opam/.opam/4.12/lib/core_kernel -I /home/opam/.opam/4.12/lib/core_kernel/base_for_tests -I /home/opam/.opam/4.12/lib/core_kernel/caml_unix -I /home/opam/.opam/4.12/lib/core_kernel/flags -I /home/opam/.opam/4.12/lib/core_kernel/version_util -I /home/opam/.opam/4.12/lib/cstruct -I /home/opam/.opam/4.12/lib/cstruct-sexp -I /home/opam/.opam/4.12/lib/domain-name -I /home/opam/.opam/4.12/lib/easy-format -I /home/opam/.opam/4.12/lib/fieldslib -I /home/opam/.opam/4.12/lib/fmt -I /home/opam/.opam/4.12/lib/gen -I /home/opam/.opam/4.12/lib/ipaddr -I /home/opam/.opam/4.12/lib/jane-street-headers -I /home/opam/.opam/4.12/lib/macaddr -I /home/opam/.opam/4.12/lib/menhirLib -I /home/opam/.opam/4.12/lib/ocaml-compiler-libs/common -I /home/opam/.opam/4.12/lib/ocaml-compiler-libs/shadow -I /home/opam/.opam/4.12/lib/ocaml-migrate-parsetree -I /home/opam/.opam/4.12/lib/ocaml/compiler-libs -I /home/opam/.opam/4.12/lib/ocaml/threads -I /home/opam/.opam/4.12/lib/ocamlgraph -I /home/opam/.opam/4.12/lib/open -I /home/opam/.opam/4.12/lib/parsexp -I /home/opam/.opam/4.12/lib/ppx_assert/runtime-lib -I /home/opam/.opam/4.12/lib/ppx_bench/runtime-lib -I /home/opam/.opam/4.12/lib/ppx_compare/runtime-lib -I /home/opam/.opam/4.12/lib/ppx_derivers -I /home/opam/.opam/4.12/lib/ppx_deriving/runtime -I /home/opam/.opam/4.12/lib/ppx_enumerate/runtime-lib -I /home/opam/.opam/4.12/lib/ppx_expect/collector -I /home/opam/.opam/4.12/lib/ppx_expect/common -I /home/opam/.opam/4.12/lib/ppx_expect/config -I /home/opam/.opam/4.12/lib/ppx_expect/config_types -I /home/opam/.opam/4.12/lib/ppx_hash/runtime-lib -I /home/opam/.opam/4.12/lib/ppx_here/runtime-lib -I /home/opam/.opam/4.12/lib/ppx_inline_test/config -I /home/opam/.opam/4.12/lib/ppx_inline_test/runtime-lib -I /home/opam/.opam/4.12/lib/ppx_module_timer/runtime -I /home/opam/.opam/4.12/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.12/lib/ppxlib -I /home/opam/.opam/4.12/lib/ppxlib/ast -I /home/opam/.opam/4.12/lib/ppxlib/print_diff -I /home/opam/.opam/4.12/lib/ppxlib/stdppx -I /home/opam/.opam/4.12/lib/ppxlib/traverse_builtins -I /home/opam/.opam/4.12/lib/result -I /home/opam/.opam/4.12/lib/sedlex -I /home/opam/.opam/4.12/lib/seq -I /home/opam/.opam/4.12/lib/sexplib -I /home/opam/.opam/4.12/lib/sexplib/unix -I /home/opam/.opam/4.12/lib/sexplib0 -I /home/opam/.opam/4.12/lib/spawn -I /home/opam/.opam/4.12/lib/splittable_random -I /home/opam/.opam/4.12/lib/stdio -I /home/opam/.opam/4.12/lib/stdlib-shims -I /home/opam/.opam/4.12/lib/tcpip -I /home/opam/.opam/4.12/lib/tcpip/unix -I /home/opam/.opam/4.12/lib/time_now -I /home/opam/.opam/4.12/lib/timezone -I /home/opam/.opam/4.12/lib/typerep -I /home/opam/.opam/4.12/lib/uchar -I /home/opam/.opam/4.12/lib/variantslib -I /home/opam/.opam/4.12/lib/yojson -I src/lib/kernel/.frenetic_kernel.objs/byte -I src/lib/netkat/.frenetic_netkat.objs/byte -no-alias-deps -open Frenetic_ppx -o src/syntax/.frenetic_ppx.objs/byte/frenetic_ppx__Ppx.cmo -c -impl src/syntax/ppx.pp.ml)
# File "src/syntax/ppx.ml", line 24, characters 18-46:
# 24 |   | Pexp_constant (Pconst_string (s, Some "")) ->
#                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
# Error: The constructor Pconst_string expects 3 argument(s),
#        but is applied here to 2 argument(s)
#     ocamlopt src/lib/netkat/.frenetic_netkat.objs/native/frenetic_netkat__Lexer.{cmx,o} (exit 2)
# (cd _build/default && /home/opam/.opam/4.12/bin/ocamlopt.opt -w -40 -safe-string -g -I src/lib/netkat/.frenetic_netkat.objs/byte -I src/lib/netkat/.frenetic_netkat.objs/native -I /home/opam/.opam/4.12/lib/astring -I /home/opam/.opam/4.12/lib/base -I /home/opam/.opam/4.12/lib/base/base_internalhash_types -I /home/opam/.opam/4.12/lib/base/caml -I /home/opam/.opam/4.12/lib/base/md5 -I /home/opam/.opam/4.12/lib/base/shadow_stdlib -I /home/opam/.opam/4.12/lib/base64 -I /home/opam/.opam/4.12/lib/base_bigstring -I /home/opam/.opam/4.12/lib/base_quickcheck -I /home/opam/.opam/4.12/lib/bigarray-compat -I /home/opam/.opam/4.12/lib/bin_prot -I /home/opam/.opam/4.12/lib/bin_prot/shape -I /home/opam/.opam/4.12/lib/biniou -I /home/opam/.opam/4.12/lib/bytes -I /home/opam/.opam/4.12/lib/core -I /home/opam/.opam/4.12/lib/core/error_checking_mutex -I /home/opam/.opam/4.12/lib/core_kernel -I /home/opam/.opam/4.12/lib/core_kernel/base_for_tests -I /home/opam/.opam/4.12/lib/core_kernel/caml_unix -I /home/opam/.opam/4.12/lib/core_kernel/flags -I /home/opam/.opam/4.12/lib/core_kernel/version_util -I /home/opam/.opam/4.12/lib/cstruct -I /home/opam/.opam/4.12/lib/cstruct-sexp -I /home/opam/.opam/4.12/lib/domain-name -I /home/opam/.opam/4.12/lib/easy-format -I /home/opam/.opam/4.12/lib/fieldslib -I /home/opam/.opam/4.12/lib/fmt -I /home/opam/.opam/4.12/lib/gen -I /home/opam/.opam/4.12/lib/ipaddr -I /home/opam/.opam/4.12/lib/jane-street-headers -I /home/opam/.opam/4.12/lib/macaddr -I /home/opam/.opam/4.12/lib/menhirLib -I /home/opam/.opam/4.12/lib/ocaml/compiler-libs -I /home/opam/.opam/4.12/lib/ocaml/threads -I /home/opam/.opam/4.12/lib/ocamlgraph -I /home/opam/.opam/4.12/lib/open -I /home/opam/.opam/4.12/lib/parsexp -I /home/opam/.opam/4.12/lib/ppx_assert/runtime-lib -I /home/opam/.opam/4.12/lib/ppx_bench/runtime-lib -I /home/opam/.opam/4.12/lib/ppx_compare/runtime-lib -I /home/opam/.opam/4.12/lib/ppx_deriving/runtime -I /home/opam/.opam/4.12/lib/ppx_enumerate/runtime-lib -I /home/opam/.opam/4.12/lib/ppx_expect/collector -I /home/opam/.opam/4.12/lib/ppx_expect/common -I /home/opam/.opam/4.12/lib/ppx_expect/config -I /home/opam/.opam/4.12/lib/ppx_expect/config_types -I /home/opam/.opam/4.12/lib/ppx_hash/runtime-lib -I /home/opam/.opam/4.12/lib/ppx_here/runtime-lib -I /home/opam/.opam/4.12/lib/ppx_inline_test/config -I /home/opam/.opam/4.12/lib/ppx_inline_test/runtime-lib -I /home/opam/.opam/4.12/lib/ppx_module_timer/runtime -I /home/opam/.opam/4.12/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.12/lib/result -I /home/opam/.opam/4.12/lib/sedlex -I /home/opam/.opam/4.12/lib/seq -I /home/opam/.opam/4.12/lib/sexplib -I /home/opam/.opam/4.12/lib/sexplib/unix -I /home/opam/.opam/4.12/lib/sexplib0 -I /home/opam/.opam/4.12/lib/spawn -I /home/opam/.opam/4.12/lib/splittable_random -I /home/opam/.opam/4.12/lib/stdio -I /home/opam/.opam/4.12/lib/stdlib-shims -I /home/opam/.opam/4.12/lib/tcpip -I /home/opam/.opam/4.12/lib/tcpip/unix -I /home/opam/.opam/4.12/lib/time_now -I /home/opam/.opam/4.12/lib/timezone -I /home/opam/.opam/4.12/lib/typerep -I /home/opam/.opam/4.12/lib/uchar -I /home/opam/.opam/4.12/lib/variantslib -I /home/opam/.opam/4.12/lib/yojson -I src/lib/kernel/.frenetic_kernel.objs/byte -I src/lib/kernel/.frenetic_kernel.objs/native -intf-suffix .ml -no-alias-deps -open Frenetic_netkat -o src/lib/netkat/.frenetic_netkat.objs/native/frenetic_netkat__Lexer.cmx -c -impl src/lib/netkat/Lexer.pp.ml)
# File "_none_", line 1:
# Error: Unbound value Sedlexing.__private__next_int
```

mseri commented on packages/frenetic/frenetic.5.0.5/opam:
```diff
@@ -0,0 +1,48 @@
+homepage: "http://frenetic-lang.org"
+bug-reports: "https://github.com/frenetic-lang/frenetic/issues"
+dev-repo: "git+https://github.com/frenetic-lang/frenetic.git"
+build: [
```
Can you please add the license in spdx form: https://spdx.org/licenses/

jnfoster replied:
Thanks. Fixed locally.

jnfoster commented:
> All builds are failing with [snipped...]

So sorry. I seem to have had an outdated version of `ppxlib` installed. Should know better than to fire-and-forget at 3am :-)

jnfoster commented:
Hi! I removed some fragile and hard to maintain PPX code that was rarely used, and updated dependencies in the `.opam` file. Let's see if this works...

kit-ty-kate commented on packages/frenetic/frenetic.5.0.5/opam:
```diff
@@ -0,0 +1,48 @@
+  "mparser-re" {>= "1.3"}
+  "ocamlgraph" {>= "2.0.0"}
+  "open" {>= "0.2.1"}
+  "ppxlib" {>= "0.22.0"}
```
```suggestion
```
Is ppxlib even used directly at all now?

jnfoster replied:
Nope! Removed.

kit-ty-kate commented on packages/frenetic/frenetic.5.0.5/opam:
```diff
@@ -0,0 +1,48 @@
+]
+depends: [
+  "ocaml" {build & >= "4.11.0"}
+  "dune" {build & >= "2.0"}
```
```suggestion
  "dune" {>= "2.0"}
```
Dune does not support the `build` tag. See https://github.com/ocaml/dune/issues/2147

jnfoster replied:
Fixed, thanks.

kit-ty-kate commented on packages/frenetic/frenetic.5.0.5/opam:
```diff
@@ -0,0 +1,48 @@
+  ["dune" "build" "@doc" "-p" name "-j" jobs] {with-doc}
+]
+depends: [
+  "ocaml" {build & >= "4.11.0"}
```
```suggestion
  "ocaml" {>= "4.11.0"}
```
OCaml is almost alway a normal dependency

jnfoster replied:
Fixed, thanks.

kit-ty-kate commented on packages/frenetic/frenetic.5.0.5/opam:
```diff
@@ -0,0 +1,48 @@
+bug-reports: "https://github.com/frenetic-lang/frenetic/issues"
+dev-repo: "git+https://github.com/frenetic-lang/frenetic.git"
+build: [
+  ["dune" "subst"] {pinned}
```
```suggestion
  ["dune" "subst"] {dev}
```
`dev` should be used here instead of `pinned`. See https://github.com/ocaml/dune/pull/3647

jnfoster replied:
Fixed, thanks.

kit-ty-kate commented on packages/frenetic/frenetic.5.0.5/opam:
```diff
@@ -0,0 +1,48 @@
+  "ppx_jane" {>= "0.14.0"}
+  "ppx_cstruct" {>= "5.2.0"}
+  "ppx_deriving" {>= "5.1"}
+  "sedlex" {>= "2.2" & <= "2.3"}
```
```suggestion
  "sedlex" {>= "2.2" & < "2.4"}
```
Using a strict equality sign is usually more correct (e.g. a 2.3.1 is always a possibility)

Also, just curious, what is the reason behind this restriction?

jnfoster replied:
I had the same error as FStar here: https://github.com/FStarLang/FStar/issues/2343

kit-ty-kate replied:
cc @toots ?

toots replied:
Thanks for letting me know! As fas as I can tell, `sedlex` `2.4` has a clean `mli` interface. This issue is due to an embedded mli of the module shipped with the projet that prevents it from linking with the new API: https://github.com/frenetic-lang/frenetic/blob/master/src/lib/netkat/Lexer.cppo.ml#L13

I believe that the missing functionalities for tracking source locations were added into Sedlex a while ago. I'd suggest to either update to a recent version or rename the custom module.

jnfoster replied:
Ok. I did the same hack as FStar, adding a `__private__next_int` field to our `LexBuffer` overlay... it's ugly, but at least we avoid brittle version pinning for now.

kit-ty-kate commented on packages/frenetic/frenetic.5.0.5/opam:
```diff
@@ -0,0 +1,48 @@
+  "ocamlgraph" {>= "2.0.0"}
+  "open" {>= "0.2.1"}
+  "ppxlib" {>= "0.22.0"}
+  "ppx_jane" {>= "0.14.0"}
```
```suggestion
  "ppx_jane" {>= "v0.14.0"}
```
The `v` prefix is required on Janestreet packages

jnfoster replied:
Fixed, thanks!

kit-ty-kate commented on packages/frenetic/frenetic.5.0.5/opam:
```diff
@@ -0,0 +1,48 @@
+  "ppx_cstruct" {>= "5.2.0"}
+  "ppx_deriving" {>= "5.1"}
+  "sedlex" {>= "2.2" & <= "2.3"}
+  "sexplib" {>= "0.14.0"}
```
```suggestion
  "sexplib" {>= "v0.14.0"}
```
The `v` prefix is required on Janestreet packages

jnfoster replied:
Fixed, thanks!

hannesm commented on packages/frenetic/frenetic.5.0.5/opam:
```diff
@@ -0,0 +1,47 @@
+  "dune" { >= "2.0"}
+  "odoc"
+  #########################
+  "async"  {>= "vv0.14.0" }
```
```suggestion
  "async"  {>= "v0.14.0" }
```

one `v` is enough ;)

hannesm commented on packages/frenetic/frenetic.5.0.5/opam:
```diff
@@ -0,0 +1,47 @@
+  "base64" {>= "3.5.0"}
+  "cohttp" {>= "4.0.0"}
+  "cohttp-async" {>= "4.0.0" }
+  "core"   {>= "vv0.14.0" }
```
```suggestion
  "core"   {>= "v0.14.0" }
```

see above

kit-ty-kate commented on packages/frenetic/frenetic.5.0.5/opam:
```diff
@@ -0,0 +1,47 @@
+depends: [
+  "ocaml" { >= "4.11.0"}
+  "dune" { >= "2.0"}
+  "odoc"
```
```suggestion
  "odoc" {with-doc}
```
I’m not seeing uses of odoc outside of the documentation

jnfoster commented:
🙏 You are very kind to review and repair the disorganized commits from a busy professor.

jnfoster commented:
Many of the CI builds are failing because of an MD5 sum. I created my PR with `opam publish`. Did I also need to do something else? Sorry for being a newbie...

kit-ty-kate commented:
> Many of the CI builds are failing because of an MD5 sum. I created my PR with `opam publish`. Did I also need to do something else? Sorry for being a newbie...

The 5.0.5 tag used for the archive has been force-pushed a couple of hours ago after your last change here, so the checksum is different from the one submitted by opam publish. Could you either force-push it back to the original or update the checksum in this PR?

jnfoster commented:
Seems like the CI is stalled only on the MacOS / Homebrew jobs.

```
2021-11-08 21:44.48: Waiting for resource in pool OCluster
```

Is there anything that can be done to get this unstuck? Did the project run out of resources for the week/month/year?

kit-ty-kate commented:
Yeah, the macOS runners are stuck. We can merge anyway i think. Thanks!

jnfoster commented:
Woohoo! Thank you again for patiently reviewing this and helping me get everything in order. It's great to have Frenetic building in opam again.
